fix(observability): register ObservationRegistry for dynamic MessageChannels in StreamBridge (#3033)

agustino-lim opened the pull request:
This PR addresses issue #3033 by ensuring ObservationRegistry is registered for dynamically created MessageChannel in StreamBridge

olegz commented:
Thank you for your contribution.
Can you please rebase and update your PR, as there are test failures in Rabbit binder, perhaps due few recent changes to StreamBridge

agustino-lim commented:
Thank you for your feedback! I have rebased my branch and addressed the test failures related to the Rabbit binder. Please let me know if there’s anything else that needs to be updated.
